doc/api/events.md

```diff
@@ -155,9 +155,63 @@ myEmitter.emit('error', new Error('whoops!'));
 // Prints: whoops! there was an error
 ```
 
+## AutoCatch of Promise rejection
+
+Using async functions in event handlers is problematic, because it would
+lead to `'unhandledRejection'` in case of an exception
+
+```js
+const ee = new EventEmitter();
+ee.on('something', async (value) => {
+  throw new Error('kaboom');
+});
+```
+
+The `autoCatch` option in the `EventEmitter` constructor or the global
+setting change this behavior, installing a `.catch()` handler on the
+`Promise`. This `catch` handler routes the exception to the
+[`Symbol.for('nodejs.rejection')`][rejection] event if there is a listener, or to
```

Comment:
Honestly this would require someone to return a promise from an event emitter method and add a catch method to it later which means it's not an async function.

I am entirely fine with not supporting this case or emitting an event here just because the added complexity likely isn't worth it.